server/src/routes/user.ts

```diff
@@ -0,0 +1,57 @@
+import { prisma } from "../common";
+import express from "express";
+import { asyncHandler } from "../utils/asyncHandler";
+
+export const userRoutes = express.Router();
+
+// Filter using query string in url with parameters role and category
+userRoutes.route("/").get(
+  asyncHandler(async (req, res) => {
+    let categoryFilter = (req.query.category as string) || null;
+    let roleFilter = (req.query.role as string) || null;
+    let filter: any = {};
+
+    if (categoryFilter !== null) {
+      const categoryId = parseInt(categoryFilter);
+      filter.categoryGroupId = categoryId;
+    }
+
+    if (roleFilter !== null) {
+      roleFilter = roleFilter.toUpperCase();
+      filter.role = roleFilter;
+    }
+
+    const users = await prisma.user.findMany({
+      where: filter,
+    });
+
+    let usersResponse: any = [];
+    users.forEach(user => {
+      const { token, ...userWithoutToken } = user;
+      usersResponse.push(userWithoutToken);
+    });
```

Comment:
For this, It's more efficient to do:
`users.map(({token, ...userWithoutToken}) => userWithoutToken)`
Then you can just return this

Actually, see next comment